src/org/parosproxy/paros/Constant.java

```diff
@@ -1086,18 +1087,15 @@ public static Path getPathDefaultConfigFile() {
     }
 
 	public static File getContextsDir () {
-		File f = new File(Constant.getZapHome(), USER_CONTEXTS_DIR);
-		if (! f.exists()) {
-			f.mkdirs();
-		}
-		if (f.isDirectory() && f.canWrite()) {
-			return f;
-		}
-		return Model.getSingleton().getOptionsParam().getUserDirectory();
+		return getDir(USER_CONTEXTS_DIR);
 	}
 
 	public static File getPoliciesDir () {
-		File f = new File(Constant.getZapHome(), USER_POLICIES_DIR);
+		return getDir(USER_POLICIES_DIR);
+	}
+
+	public static File getDir (String directory) {

```

Comment:
- Is it public on purpose?
 - `getHomeDir`? (`getDir` is too generic.)
 - `directory` → `child` or `childDir`.

Reply:
added a commit to resolved the code-review comment, @thc202 please review.

Reply:
- The method is still public(?).
 - I'd prefer a name that contained "home" instead (iirc we never referred to the directories in the home dir as a user directories, despite the constant names). In any case the PR title, commit message, and ZAP comment needs to be updated (or reworded to not include the new method e.g. "Reduce code duplication in getContextsDir and getPoliciesDir").

Reply:
@thc202 thank you, resolved the code-review comments.